src/app/app.module.ts

```diff
@@ -20,14 +27,25 @@ import { BasicComponent } from './templates/basic/basic.component';
     SignupComponent,
     StudentComponent,
     OrganizationComponent,
-    BasicComponent
+    BasicComponent,
+    FooterComponent,
+    NavbarComponent,
   ],
   imports: [
     BrowserModule,
     AppRoutingModule,
-    AgGridModule.withComponents([])
+    AgGridModule.withComponents([]),
+    JwtModule.forRoot({
+      config:{
+        tokenGetter: () => {
+          return localStorage.getItem('token');
+        }
```

Comment:
I would recommend you try observables to implement guards

Reply:
The angular-jwt library doesn't support Observables, but it does support Promises, so I've used that instead

Reply:
there's no need of JWT. authentication can be done using observables alone. Can you try that if you have time

Reply:
Is it just like checking if the token exists

Reply:
yes

Reply:
but you no need to even store it in localStorage as it may not be the safest option. I recommend you to see for the difference between observables and promises.

Reply:
So the authentication guard serves no security purpose whatsoever. What about limiting access to organization route?

Reply:
I guess there's a confusion. Implementing auth guards (restricting access to routes) can be done using observables alone without even need of storing the token in local storage

Reply:
If I don't store the token, how would I know if the user is authenticated when they navigate to another route?

Reply:
that's why we use observables, please give it a read, you might want to check this https://stackoverflow.com/questions/37948068/angular-2-routing-canactivate-work-with-observable

Reply:
What I'm unsure of is how the `loginService.isLoggedIn()` works

Reply:
Hmm maybe it would be possible to send a request to `/api/get_certificates` to check if logged in.

Reply:
But the heroku endpoint still blocks localhost

Reply:
isLoggedIn():Observable<boolean> {
//this is where you first try to get user info from backend {/api/getUserInfo} and send info ahead if present
    return this.http.get(ApiResources.LOGON).map(response => response.ok);
}

Reply:
Yes that would be possible. Is it possible to unblock localhost?

Reply:
I've changed the code to use the `/api/get_certificates` endpoint for authentication and observables